agent/doc/swdesign/README.md

```diff
@@ -1315,6 +1336,101 @@ Needs:
 - impl
 - utest
 
+#### WorkloadControlLoop handles restarts of workloads
+`swdd~workload-control-loop-handles-workload-restarts~1`
+
+Status: approved
+
+When the WorkloadControlLoop receives a new workload state, then the WorkloadControlLoop shall:
+- triggers the check of the permissibility of the restart
+- execute the restart of the workload it manages based on the result of the comparison
+
+Rationale:
+The execution of a restart of the workload depends on the workload state and the configured restart policy.
+
+Tags:
+- WorkloadControlLoop
+
+Needs:
+- impl
+- utest
+
+#### WorkloadControlLoop restarts workload with enabled restart policy
+`swdd~workload-control-loop-restarts-workload-with-enabled-restart-policy~1`
+
+Status: approved
+
+When:
+the workload's `ExecutionState` is `Succeeded(Ok)` or `ExecutionState` is `Failed(ExecFailed)` and the workload's configured `RestartPolicy` contains the value `ALWAYS` or
+the workload's `ExecutionState` is `Failed(ExecFailed)` and the workload's configured `RestartPolicy` contains the value `ON_FAILURE`,
+then the WorkloadControlLoop shall restart the workload.
+
+Rationale:
+The restart depends on the execution state of the workload.
+
+Tags:
+- WorkloadControlLoop
+
+Needs:
+- impl
+- utest
+- stest
+
+#### WorkloadControlLoop does not restart a workload with disabled restart policy
+`swdd~workload-control-loop-no-restart-with-disabled-restart-policy~1`
+
+Status: approved
+
+When the workload's configured `RestartPolicy` contains the value `NEVER`, then the WorkloadControlLoop shall not restart the workload.
+
+Rationale:
+The user has explicitly configured the workload to prevent its restart or to apply the default value of "NEVER" in the workload configuration.
+
+Tags:
+- WorkloadControlLoop
+
+Needs:
+- impl
+- utest
+- stest
+
+#### WorkloadControlLoop restarts workloads using the update operation
+`swdd~workload-control-loop-restarts-workloads-using-update~1`
+
+Status: approved
+
+When the WorkloadControlLoop executes restarts of the workload, then the WorkloadControlLoop shall:
+- delete the existing workload via the corresponding runtime connector
+- create a new workload with the stored workload configuration via the corresponding runtime connector
+
+Comment:
+The restart is represented within the system by an update operation.
+
+Rationale:
+A runtime may not support directly restarting the exited container.
+
+Tags:
+- WorkloadControlLoop
+
+Needs:
+- impl
+- utest
+
+#### WorkloadControlLoop skips restarts
+`swdd~workload-control-loop-skips-restarts~1`
+
+When the WorkloadControlLoop handles a restart of a workload and the workload's current `WorkloadInstanceName` inside the WorkloadControlLoop differs from the `WorkloadInstanceName` inside the received workload state, then the WorkloadControlLoop shall skip the restart.
```

Comment:
We have 3 requirements about the preconditions of the restart, but they are somehow not really connected together.
Maybe we should mention this here in the requirement above and reorder a bit:

1. this requirement here with a comment like "If the workload changed in the meantime, there is no need to restart it."
2. one requirement about the execution states which states "If the worklaod has not changed and either the workload's `ExecutionS.... or .... the workload's `ExecutionState` is `Failed ... then"

Reply:
I have fixed it.